jellyfin-api/src/main/kotlin/org/jellyfin/sdk/api/client/KtorClient.kt

```diff
@@ -44,29 +43,76 @@ public open class KtorClient(
 		}
 	}
 
-	override fun createPath(path: String, pathParameters: Map<String, Any?>): String = path
-		.split('/')
-		.filterNot { it.isEmpty() }
-		.mapNotNull { rawName ->
-			val name = rawName.removeSurrounding(prefix = "{", suffix = "}")
+	override fun createPath(path: String, pathParameters: Map<String, Any?>): String {
+		val result = StringBuilder()
 
-			// Only act if the name actually is a variable
-			if (name != rawName) {
-				// Check if key is set
-				if (!pathParameters.containsKey(name))
-					throw MissingPathVariableError(name, path)
+		var lastStart = -1
+		var lastEnd = -1
+
+		for (i in path.indices) {
+			when (path[i]) {
+				'/' -> {
+					check(lastStart < 0) {
+						"Unclosed path variable ${path.substring(lastStart, i)} in path $path"
+					}
+
+					// Append content from last path variable or slash up to here, eliminating duplicate slashes
+					val content = path.substring(lastEnd + 1, i + 1)
+					if (content != "/" || (result.isNotEmpty() && result.lastOrNull() != '/')) {
+						result.append(content)
+					}
+
+					lastEnd = i
+				}
+				'{' -> {
+					check(lastStart < 0) {
+						"Nested path variable at $i in path $path"
+					}
+
+					// Append content from last path variable end up to here
+					result.append(path.substring(lastEnd + 1, i))
+
+					// Set path variable start index (exclude opening brace)
+					lastStart = i + 1
+				}
+				'}' -> {
+					check(lastStart >= 0) {
+						"End of path variable without start at $i in path $path"
+					}
+					lastEnd = i
+
+					// Get path variable key
+					val name = path.substring(lastStart, lastEnd)
+
+					// Check if key is set
+					if (!pathParameters.containsKey(name)) {
+						throw MissingPathVariableError(name, path)
+					}
```

Comment:
```suggestion
					if (!pathParameters.containsKey(name)) throw MissingPathVariableError(name, path)
```

Reply:
The line was a bit long that way, so I prefer putting it below.

Reply:
```suggestion
					if (!pathParameters.containsKey(name))
						throw MissingPathVariableError(name, path)
```

Reply:
It seems since it's still below the 120 char line limit, I can also put it in the same line.